Add section “Maintaining etcd clusters” into “Operating etcd clusters for Kubernetes” (#40817)
kubernetes/website: content/en/docs/tasks/administer-cluster/configure-upgrade-etcd.md

{{< note >}}
Before you start an upgrade, please back up your etcd cluster first.
{{< /note >}}

## Maintaining etcd clusters

Fore more details on etcd maintenance, please refer to the [etcd maintenance](https://etcd.io/docs/latest/op-guide/maintenance/) documentation.

{{< note >}}
Defragmentation is an expensive operation, so it should be executed as infrequent
as possible. On the other hand, it's also necessary to make sure any etcd member
will not run out of the storage quota. It's recommended to run defragmentation

> using [etcd-defrag](https://github.com/ahrtr/etcd-defrag), which is an easier to

Reviewer: i think the sig docs rules had something about a third party shortcode / disclaimer that has to be added here, but i don't remember the details.

Reviewer 2: In order of preference:
1. do not make a specific recommendation [following [https://kubernetes.io/blog/2020/05/third-party-dual-sourced-content/](https://kubernetes.io/blog/2020/05/third-party-dual-sourced-content/)] - etcd is not part of the K8s project
2. link to external docs that recommend the tool.
3. add a third party content disclaimer shortcode

Please pick one of these.

Until then:
/lgtm cancel

(fine to LGTM it back once that's addressed)

Reviewer: > 3\. add a third party content disclaimer shortcode

the shortcode was:
`{{% thirdparty-content %}}`

Author: Thanks both & updated.

Author: @neolit123 @sftim PTAL, thx

{{< note >}}
Before you start an upgrade, please back up your etcd cluster first.
{{< /note >}}

## Maintaining etcd clusters

Fore more details on etcd maintenance, please refer to the [etcd maintenance](https://etcd.io/docs/latest/op-guide/maintenance/) documentation.

> {{% thirdparty-content %}}

Reviewer 2: (nit, but good to fix)
```suggestion
{{% thirdparty-content single="true" %}}
```

Author: thx & updated.

{{< note >}}
Defragmentation is an expensive operation, so it should be executed as infrequent
as possible. On the other hand, it's also necessary to make sure any etcd member

> will not run out of the storage quota. It's recommended to run defragmentation
> using [etcd-defrag](https://github.com/ahrtr/etcd-defrag), which is an easier to
> use and smarter etcd defragmentation tool.

Reviewer 2: ```suggestion
will not run out of the storage quota. The Kubernetes project recommends that when you
perform defragmentation, you use a tool such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag).
```
or
```suggestion
will not run out of the storage quota. The Kubernetes project recommends that when you
perform defragmentation, you use a tool such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag).
Compared to the tooling provided by the etcd project itself, `etcd-defrag` is easier to use and smarter.
```

Ideally, avoid this kind of comparative evaluation. We try to avoid making critiques between different third party tools, as these are an extra effort to maintain. It's hard to find Kubernetes contributors who can keep that kind of information current.

Author: thx & updated. PTAL
